pkg/ocm/storage/outcoming/ocm.go

```diff
@@ -396,14 +397,16 @@ func getUploadProtocol(protocols []*gateway.FileUploadProtocol, protocol string)
 	return "", "", false
 }
 
-func (d *driver) Upload(ctx context.Context, ref *provider.Reference, content io.ReadCloser) error {
+func (d *driver) Upload(ctx context.Context, ref *provider.Reference, content io.ReadCloser, metadata map[string]string) error {
 	share, rel, err := d.shareAndRelativePathFromRef(ctx, ref)
 	if err != nil {
 		return err
 	}
```

Comment:
A check is needed to make sure `metadata` is not nil to avoid runtime errors (nil pointers) down the code.

Reply:
But `metadata` here is an implicit reference as it's a `map`, so cannot be nil right?